cpp/src/arrow/flight/meson.build

```diff
@@ -44,7 +44,17 @@ install_headers(
     subdir: 'arrow/flight',
 )
 
-grpc_dep = dependency('grpc++')
+# We specifically need to use the grpc subproject and cannot use a system installion
+# due to https://github.com/mesonbuild/meson/issues/14905
```

Comment:
Can we use https://mesonbuild.com/Reference-manual_returned_external_program.html#external_programfull_path as workaround?
If we can use `grpc_cpp_plugin.full_path()`, can we keep supporting system gRPC?

Reply:
Oh... I see.